src/Umbraco.Web/Editors/IconController.cs

```diff
@@ -34,5 +41,18 @@ public IList<IconModel> GetAllIcons()
         {
             return _iconService.GetAllIcons();
         }
+
+        /// <summary>
+        /// Gets JSON blob containing all the known icons
+        /// </summary>
+        /// <param name="culture"></param>
+        /// <returns></returns>
+        public JsonNetResult GetIcons()
+        {
+            var dictionary = _iconService.GetAllIcons()
+                .ToDictionary(i => i.Name, i => i.SvgString);      
+
+            return new JsonNetResult { Data = dictionary, Formatting = Formatting.None };
```

Comment:
JsonNetResult currently creates a new JsonSerializerSettings instance on construction, so this needs https://github.com/umbraco/Umbraco-CMS/pull/9670 merged to avoid this being slower.